prow/cmd/tide/README.md

```diff
@@ -24,7 +24,96 @@ them when they have up-to-date passing test results ("tide goes out").
+            end
+        end
+
+        subgraph prow-cluster
+            prowjobs[Prowjobs];
+            config.yaml;
+        end
+
+        subgraph tide-workflow
+            Tide;
+            pools;
+            filtered-pools
+            subgraph sync-sub-pool
+                single-pool;
+                pool-n;
+                pool-n1;
+                accumulated-batch-prowjobs-->|filter out <br> incorrect refs <br> no longer meet merge requirement|valid-batches;
+                valid-batches-->accumulated-batch-success;
+                valid-batches-->accumulated-batch-pending;
+                
+                status-context-->filtered-prowjobs;
+                filtered-prowjobs-->|accumulate|map_context_best-result;
+                map_context_best-result-->map_pr_overall-results;
+                map_pr_overall-results-->accumulated-success;
+                map_pr_overall-results-->accumulated-pending;
+                map_pr_overall-results-->accumulated-stale;
+                
+                subgraph all-accumulated-pools
+                    accumulated-batch-success;
+                    accumulated-batch-pending;
+                    accumulated-success;
+                    accumulated-pending;
+                    accumulated-stale;
+                end
+
+                accumulated-batch-success-..->accumulated-batch-success-exist{Exist};
+                accumulated-batch-pending-..->accumulated-batch-pending-exist{Exist};
+                accumulated-success-..->accumulated-success-exist{Exist};
+                accumulated-pending-..->accumulated-pending-exist{Exist};
+                accumulated-stale-..->accumulated-stale-exist{Exist};
+                accumulated-batch-success-exist-->|yes|merge-batch[Merge batch];
+                accumulated-batch-success-exist-->|no|accumulated-batch-pending-exist;
+                accumulated-batch-pending-exist-->|no|accumulated-success-exist;
+                accumulated-success-exist-->|yes|merge-single[Merge Single];
+                accumulated-success-exist-->|no|accumulated-pending-exist;
+                accumulated-pending-exist-->|no|can-trigger-batch{Can Trigger New Batch};
+                can-trigger-batch-->|yes|trigger-batch[Trigger new batch];
+                can-trigger-batch-->|no|accumulated-stale-exist;
+                accumulated-stale-exist-->|yes|trigger-highest-pr[Trigger Jobs on Highest Priority PR];
+            end
+        end
+
+        Tide-->pools;
+        pullrequest-->pools[Pools - grouped PRs, prow jobs by org/repo/branch];
+        pools-->|filter out prs <br> failed prow jobs <br> pending non prow checks <br> merge conflict <br> invalid merge method|filtered-pools;
+        head-ref-->pools;
+        prowjobs-->pools;
+        config.yaml-->pools;
+        filtered-pools-->single-pool;
+        filtered-pools-->pool-n;
+        filtered-pools-->pool-n1[pool ...];
+        single-pool-->status-context;
```

Comment:
I don't understand this?

Reply:
`single-pool` might be clearer and more consistent with the other pool names as `pool-i`.

The reason it links to `status-context` is that we use status contexts to construct synthetic PJ resources in case the original PJs were garbage collected. It would probably make more sense for this arrow to start from `accumulate-serial` instead.

Reply:
Ah okay, I thought it was maybe somehow referring to the fallback of getting contexts via v3 api if we didn't get them via v4...We should probably have that explanation on the arrow somehow if possible

Reply:
added explanation on the line